### DISCUSSION TITLE
/inst/extdata folder

### DISCUSSION THREAD

**Discussion by eva0marques** [Spatiotemporal-Exposures-and-Toxicology/NRT-AP-Model#222] [Show and tell]
@Spatiotemporal-Exposures-and-Toxicology @dawranadeep @MAKassien @dzilber @sigmafelix @Sanisha003 
I created the folder /inst/extdata (see https://r-pkgs.org/data.html) that is supposed to store all data created and used in our package. For example, I stored a table with NLCD classes names, abbreviation, meaning and associated colors in nlcd_classes.csv. It is used for renaming nlcd ratio in my .R function.

**Comment by kyle-messier** [comment]
@eva0marques  This is an interesting solution. I am not seeing the documentation mention how it deals with data size and storage on GitHub.

**Reply by eva0marques** [reply]
Yes, I don't know. It is more for small data I guess, like dictionary or small csv files that need to be used within .R functions. It is not for the input or output data. It was @sigmafelix idea.

**Reply by sigmafelix** [reply]
All packages submitted to CRAN should abide the size limit of <=10 MB in tarball size and <=5 MB in documentation/data ([cf. 10th bullet](https://cran.r-project.org/web/packages/policies.html#Source-packages)). I think we need to reduce the data size to as small as possible since we aim to submit this package to CRAN. Large data-only packages are usually hosted in non-CRAN locations (e.g., [spDataLarge package](https://github.com/Nowosad/spDataLarge)). Our study has a different situation where dataset sizes often exceed several gigabytes. I supposed this would be a fundamental question about the purpose of NRTAPModel package: do we want to share the full workflow with data for full reproducibility or want to make the pipeline be able to be maintained with CI/TDD? If the former is our ultimate goal, all raw datasets need to be stored elsewhere on the internet; however, if the objective is the latter, I think that sharing our pipeline, fully tested functions, and regularly maintained raw data sources (e.g., URLs) without _all_ raw datasets might be fine. Users will be able to download the entire raw datasets with the functions if they want to reproduce our results.
